packages/babel-traverse/test/modification.js

```diff
@@ -335,16 +335,65 @@ describe("modification", function () {
         );
       });
 
-      it("the exported expression", function () {
-        const bodyPath = getPath("export default fn();", {
-          sourceType: "module",
-        }).parentPath;
-        const path = bodyPath.get("body.0.declaration");
-        path.insertAfter(t.identifier("x"));
+      it("edge case for babel-plugin-jest-hoist", function () {
+        let varsHoistPoint;
+        let i = 0;
+        const logs = [[], []];
+        traverse(parse(`{a;}`), {
+          BlockStatement(path) {
+            [varsHoistPoint] = path.unshiftContainer(
+              "body",
+              t.emptyStatement(),
+            );
+            path.traverse({
+              Identifier(path) {
+                if (i++ >= 5) {
+                  return;
+                }
+                logs[0].push(path.node.name);
+                varsHoistPoint.insertBefore(t.identifier("b"));
+              },
+            });
+          },
+        });
+        i = 0;
+        traverse(parse(`{a;}`), {
+          BlockStatement(path) {
+            [varsHoistPoint] = path.unshiftContainer(
+              "body",
+              t.emptyStatement(),
+            );
+            path.traverse({
+              enter() {},
+              Identifier(path) {
+                if (i++ >= 5) {
+                  return;
+                }
+                logs[1].push(path.node.name);
+                varsHoistPoint.insertBefore(t.identifier("b"));
+              },
+            });
+          },
+        });
 
-        expect(generateCode({ parentPath: bodyPath })).toBe(
-          "var _temp;\nexport default (_temp = fn(), x, _temp);",
-        );
+        expect(logs).toMatchInlineSnapshot(`
+          Array [
+            Array [
+              "a",
+              "b",
+              "b",
+              "b",
+              "b",
+            ],
+            Array [
+              "a",
+              "b",
+              "b",
+              "b",
+              "b",
+            ],
+          ]
```

Comment:
Technically Jest's CI failure is not a bug.
The new behavior seems to make more sense than the old one.
Previously we didn't traverse the `emptyStatement` due to https://github.com/babel/babel/blob/a6aa8a900bb3388edb636874a80e6f985d9dd819/packages/babel-traverse/src/context.ts#L34-L56 https://github.com/babel/babel/blob/a6aa8a900bb3388edb636874a80e6f985d9dd819/packages/babel-traverse/src/context.ts#L92-L99, so it had no `contexts` and the queue was not added correctly.

Old snapshots
```
expect(logs).toMatchInlineSnapshot(`
  Array [
    Array [
      "a",
    ],
    Array [
      "a",
      "b",
      "b",
      "b",
      "b",
    ],
  ]
`);
```

Of course we should avoid this change in non-major releases anyway, because it will affect Jest users.